cg/meta/encryption/encryption.py

```diff
@@ -92,6 +131,160 @@ def get_symmetric_decryption_command(
         decryption_parameters.extend(output_parameter)
         return decryption_parameters
 
+    def create_pending_file(self, pending_path: Path) -> None:
+        """Create a pending flag file."""
+        LOG.info(f"Creating pending flag {pending_path}")
+        if self.dry_run:
+            return
+        pending_path.touch(exist_ok=False)
+
+
+class FlowCellEncryptionAPI(EncryptionAPI):
+    """Encryption functionality for flow cells."""
+
+    def __init__(
+        self,
+        binary_path: str,
+        encryption_dir: Path,
+        flow_cell: FlowCellDirectoryData,
+        pigz_binary_path: str,
+        sbatch_parameter: Dict[str, Union[str, int]],
+        slurm_api: SlurmAPI,
+        tar_api: TarAPI,
+        dry_run: bool = False,
+    ):
+        super().__init__(binary_path=binary_path, dry_run=dry_run)
+        self.flow_cell: FlowCellDirectoryData = flow_cell
+        self.encryption_dir: Path = encryption_dir
+        self.tar_api: TarAPI = tar_api
+        self.pigz_binary_path: str = pigz_binary_path
+        self.slurm_api: SlurmAPI = slurm_api
+        self.slurm_account: str = sbatch_parameter.get("account")
+        self.slurm_hours: int = sbatch_parameter.get("hours")
+        self.slurm_mail_user: str = sbatch_parameter.get("mail_user")
+        self.slurm_memory: int = sbatch_parameter.get("memory")
+        self.slurm_number_tasks: int = sbatch_parameter.get("number_tasks")
+        self.slurm_api.set_dry_run(dry_run=self.dry_run)
+
+    @property
+    def flow_cell_encryption_dir(self) -> Path:
+        return Path(self.encryption_dir, self.flow_cell.full_name)
+
+    @property
+    def flow_cell_encrypt_file_path_prefix(self) -> Path:
+        return Path(self.flow_cell_encryption_dir, self.flow_cell.id)
+
+    @property
+    def pending_file_path(self) -> Path:
+        return Path(self.flow_cell_encrypt_file_path_prefix.with_suffix(FileExtensions.PENDING))
+
+    @property
+    def complete_file_path(self) -> Path:
+        return Path(self.flow_cell_encrypt_file_path_prefix.with_suffix(FileExtensions.COMPLETE))
```

Comment:
Not for this PR but just as a note. with the updated proposed `FlowCellDirectoryData` refactoring these attributes would be part of the `EncryptFlowCellData` child class. So instead of `FlowCellDirectoryData` this API would use the `EncryptFlowCellData` that contains these properties.